lib/travis/build/appliances.rb

```diff
@@ -55,14 +55,23 @@
 module Travis
   module Build
     module Appliances
+      attr_reader :app
+
       def apply(name)
-        app = appliance(name)
-        app.apply if app.apply?
+        @app = appliance(name)
+        with_timer(name) { app.apply } if app.apply?
       end
 
       def appliance(name)
         Appliances.const_get(name.to_s.camelize).new(self)
       end
+
+      def with_timer(name)
```

Comment:
We probably want to move this to `Travis::Shell::Builder` if we allow timers to be stacked.